### PR TITLE
Feature/search meme

### PR DESCRIPTION
**Pull request by Vlad-Gheorghiu**
Implemented **Search meme** functionality through a search bar at top of the site. When on Posts page there is a live search by filtering the memes. While not on Posts page the live search is disabled. Also, the suggestion drop down menu is updated with its results (selecting one will take you the meme's details page)

### CODE REVIEW

File: src/app/core/components/root/app.component.scss [hunk 1]
```diff
@@ -9,4 +9,10 @@
 .toolbar-action {
     padding: 0 14px;
     margin: 0 8px;
+}
+
+#search-bar {
+    font-size: 16px;
+    width: 100%;
+    margin: 0 8px;
 }
```
Comment (lolismatic) [inline src/app/core/components/root/app.component.scss]: Missing eof

File: src/app/core/components/root/app.component.ts [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
+      .pipe(
+        debounceTime(500),
+        switchMap(title => this.postsService.searchPosts(title))
+      );
+  }
+
+  searchPosts(title) {
+    if (title !== null) {
+      this.searchPostEventEmitter.emit(title);
+      if (this.router.url.endsWith('/posts') || this.router.url.includes('/posts?q=')) {
```
Comment (lolismatic) [inline src/app/core/components/root/app.component.ts]: Use ActivatedRoute aR.snapshot.queryParams.q to check if q exists, also aR.snapshot.path to check the path.

File: src/app/core/components/root/app.component.scss (hunk 1 above)
Comment (Vlad-Gheorghiu) [reply src/app/core/components/root/app.component.scss]: Added eof.

File: src/app/core/components/root/app.component.ts (hunk 2 above)
Comment (Vlad-Gheorghiu) [reply src/app/core/components/root/app.component.ts]: Modified to using activatedRoute.snapshot

**Pull request merged**